### PR TITLE
feat: add permissions to share spaces

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Since share spaces are virtual, they and their permissions can't be retrieved from the server. Instead, we have to load the permissions via the underlying share. This change does exactly that and persists the permissions for the current user within the space object so they can be accessed and checked when needed.

Also makes sure that share spaces are being added to the spaces store so they're not just temporary objects.

## Related Issue

- Works towards https://github.com/owncloud/web/issues/11360

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-app-files/src/services/folder/loaderSpace.ts [hunk 1]
```diff
@@ -33,10 +44,13 @@ export class FolderLoaderSpace implements FolderLoader {
   }
 
   public getTask(context: TaskContext): FolderLoaderTask {
-    const { router, clientService, resourcesStore, userStore } = context
-    const { webdav } = clientService
+    const { router, clientService, resourcesStore, userStore, spacesStore, sharesStore } = context
+    const { webdav, graphAuthenticated: graphClient } = clientService
```
Comment (kulmann) [inline packages/web-app-files/src/services/folder/loaderSpace.ts:48]: Didn't we have the issue of the different clients losing the access token reactivity when destructuring the clientService? 🤔

Comment (JammingBen) [reply packages/web-app-files/src/services/folder/loaderSpace.ts:48]: This function gets called every time the loader performs, so the token would need to expire within the milliseconds between here and the function call 😅

File: packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts [hunk 2]
```diff
@@ -63,12 +62,15 @@ export const useGetMatchingSpace = (options?: GetMatchingSpaceOptions) => {
       shareName = resource.name
     }
 
-    return buildShareSpaceResource({
-      driveAliasPrefix,
-      id: resource.remoteItemId,
-      shareName,
-      serverUrl: configStore.serverUrl
-    })
+    const shareSpace =
```
Comment (kulmann) [inline packages/web-pkg/src/composables/spaces/useGetMatchingSpace.ts:65]: can be an inline return

File: packages/web-app-files/src/services/folder/loaderSpace.ts [hunk 3]
```diff
@@ -102,4 +132,77 @@ export class FolderLoaderSpace implements FolderLoader {
...
+    }
+
+    const allPermissions: string[] = []
+    permissions.forEach((permission) => {
+      if (permission['@libre.graph.permissions.actions']) {
+        allPermissions.push(...permission['@libre.graph.permissions.actions'])
+        return
+      }
+      const role = sharesStore.graphRoles[permission.roles[0]]
+      if (role) {
```
Comment (AlexAndBear) [inline packages/web-app-files/src/services/folder/loaderSpace.ts:190]: rather have an early return here, which translates to 'continue' in a regular for each

**Review by AlexAndBear — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/services/folder/loaderSpace.ts (hunk 3 above)
Comment (JammingBen) [reply packages/web-app-files/src/services/folder/loaderSpace.ts:190]: Why? 😅 The if-block has 2 simple lines of code, so it wouldn't enhance readability or save up anything.

Comment (AlexAndBear) [reply packages/web-app-files/src/services/folder/loaderSpace.ts:190]: For me it indeed improves readability 🤭

**Review by AlexAndBear — APPROVED** [review]

**Pull request merged**